src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -59,6 +60,9 @@ public SemanticModel(Compilation compilation, BicepFile sourceFile, IFileResolve
             // and configuration loading until the linter is actually needed
             this.linterAnalyzerLazy = new Lazy<LinterAnalyzer>(() => new LinterAnalyzer());
 
+            // lazy load single use diagnostic set
+            this.allDiagnostics = new Lazy<IEnumerable<IDiagnostic>>(() => AssembleDiagnostics(default));
```

Comment:
>allDiagnostics [](http://example.com/codeflow?start=17&length=14)

No concern with caching the diagnostics (minus the comment below), but I'm not following how this fixes the perf issue.

The expected VS code flow is to recompile on every bicep document change and then grab all the diagnostics and publish them back to the client.

Is that not true anymore? Is something else trying to collect diagnostics repeatedly?

Reply:
@jfleisher explained over email that the slowdown was due to repeated calls to `HasErrors()`.